blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/PipelineSearch.java

```diff
@@ -107,11 +108,21 @@ public String getType() {
             });
         }else{
             GlobMatcher matcher = pipeline.contains("*") ? new GlobMatcher(pipeline) : null;
-            while (pipelineIterator.hasNext()) {
-                BluePipeline p = pipelineIterator.next();
-                // If using glob syntax try to match using the glob matcher otherwise fall back to equality check
-                if (matcher != null && matcher.matches(p.getFullName())) {
-                    pipelines.add(p);
+            if (matcher != null) {
+                while (pipelineIterator.hasNext()) {
+                    BluePipeline p = pipelineIterator.next();
+                    String decodedName = URLDecoder.decode(p.getFullDisplayName());
```

Comment:
p.getFullDisplayName() is individual job/folder's encoded display name separated by '/'.  For example if there is branch with display name `feature/x-1` inside org folder `vivek`. It will appear as `vivek/feature%2Fx-1`. url decoded form will be `vivek/feature/x-1`. Just making sure query is sent accordingly. 

Also, better to use `URLDecoder.decode(param)` is deprecated, use the one that takes encoding in addition, "UTF-8"?